libs/shared/src/consts/handlebar-helpers/getTemplateVariables.ts

```diff
@@ -0,0 +1,91 @@
+import { TemplateVariableTypeEnum } from '../../entities/message-template';
+import { HandlebarHelpersEnum } from './handlebarHelpers';
+
+export interface IMustacheVariable {
+  type: TemplateVariableTypeEnum;
+  name: string;
+  defaultValue?: string | boolean;
+  required?: boolean;
+}
+
+export function getTemplateVariables(bod: any[]): IMustacheVariable[] {
+  const stringVariables: IMustacheVariable[] = bod
+    .filter((body) => body.type === 'MustacheStatement')
+    .flatMap((body) => {
+      const varName = body.params[0]?.original || (body.path.original as string);
+      if (!shouldAddVariable(varName)) {
+        return [];
+      }
+
+      if (body.params[0]?.original) {
+        if (!(Object.values(HandlebarHelpersEnum) as string[]).includes(body.path.original)) {
+          return [];
+        }
+      }
+
+      return {
+        type: TemplateVariableTypeEnum.STRING,
+        name: body.params[0]?.original || (body.path.original as string),
+        defaultValue: '',
+        required: false,
+      };
+    });
+
+  const arrayVariables: IMustacheVariable[] = bod
+    .filter((body) => body.type === 'BlockStatement' && ['each', 'with'].includes(body.path.original))
+    .flatMap((body) => {
+      const varName = body.params[0]?.original || (body.path.original as string);
+      if (!shouldAddVariable(varName)) {
+        return [];
+      }
+
+      const nestedVariablesInBlock = getTemplateVariables(body.program.body).map((mustVar) => {
+        return {
+          ...mustVar,
+          name: `${varName}.${mustVar.name}`,
+        };
+      });
+
+      return [
+        {
+          type: TemplateVariableTypeEnum.ARRAY,
+          name: varName,
+          required: false,
+        },
+        ...nestedVariablesInBlock,
+      ];
+    });
+
+  const boolVariables: IMustacheVariable[] = bod
+    .filter((body) => body.type === 'BlockStatement' && ['if', 'unless'].includes(body.path.original))
```

Comment:
added support for the [#unless block helper](https://docs.celigo.com/hc/en-us/articles/360039326071-Handlebars-helper-reference#unless)